src/protocol_handle_stream.rs

```diff
@@ -285,36 +287,37 @@ where
             return Poll::Ready(None);
         }
 
-        loop {
-            match Pin::new(&mut self.receiver).as_mut().poll_next(cx) {
-                Poll::Ready(Some(event)) => self.handle_event(event),
-                Poll::Ready(None) => {
-                    debug!(
-                        "ServiceProtocolStream({:?}) finished",
-                        self.handle_context.proto_id
-                    );
-                    self.current_task.idle();
-                    return Poll::Ready(None);
-                }
-                Poll::Pending => {
-                    break;
-                }
+        let mut is_pending = false;
+
+        match Pin::new(&mut self.receiver).as_mut().poll_next(cx) {
+            Poll::Ready(Some(event)) => self.handle_event(event),
+            Poll::Ready(None) => {
+                debug!(
+                    "ServiceProtocolStream({:?}) finished",
+                    self.handle_context.proto_id
+                );
+                self.current_task.idle();
+                return Poll::Ready(None);
```

Comment:
Better put a `is_pending &= false` here for consist. Otherwise, if future change move this `match` block to bottom will case bug.